src/react/HomePage.js

```diff
@@ -29,185 +29,166 @@ import {
-      .then(this.updateTriggerNames)
-      .catch((e) => this.handleSectionError("triggers", e));
+      } catch (err) {
+        setStatus("err");
+      }
+    };
 
-    this.astarte.getRealmManagementHealth()
-      .then(() => { this.updateStatus("realmManagementStatus", "ok") })
-      .catch(() => { this.updateStatus("realmManagementStatus", "err") });
+    getEndpointHealth();
+    return (() => { ignore = true; });
+  }, [queryAstarteHealth]);
 
-    this.astarte.getAppengineHealth()
-      .then(() => { this.updateStatus("appengineStatus", "ok") })
-      .catch(() => { this.updateStatus("appengineStatus", "err") });
+  return status;
+}
 
-    this.astarte.getPairingHealth()
-      .then(() => { this.updateStatus("pairingStatus", "ok") })
-      .catch(() => { this.updateStatus("pairingStatus", "err") });
+const useAstarteAPI = (fetchData) => {
+  const [data, setData] = useState({});
+  const [status, setStatus] = useState("loading");
+  const [error, setError] = useState({});
 
-    if (this.queryFlowHealth) {
-      this.astarte.getFlowHealth()
-        .then(() => { this.updateStatus("flowStatus", "ok") })
-        .catch(() => { this.updateStatus("flowStatus", "err") });
+  useEffect(() => {
+    if (!fetchData) {
+      setStatus("err");
+      setError({ message: "Invalid fetch method" });
+      return;
     }
-  }
 
-  updateDeviceStats(response) {
-    this.setState({
-      connectedDevices: response.data.connected_devices,
-      totalDevices: response.data.total_devices
-    });
-  }
+    let ignore = false;
 
-  updateInterfaceNames(response) {
-    this.setState({
-      interfaces: response.data
-    });
-  }
+    const getData = async () => {
+      try {
+        const response = await fetchData();
+        if (!ignore) {
+          setData(response.data);
+          setStatus("ok");
+        }
 
-  updateTriggerNames(response) {
-    this.setState({
-      triggers: response.data
-    });
-  }
+      } catch (err) {
+        setError(err);
+        setStatus("err");
 
-  handleSectionError(section, err) {
-    this.setState((state) => {
-      return (state[section] === "loading" ? {[section]: "err"} : state)
-    });
+      }
+    };
+
+    getData();
+    return (() => { ignore = true; });
+  }, [fetchData]);
+
+  return [ data, status, error ];
+}
+
+export default ({ astarte, history }) => {
+  const [ devicesStats, deviceStatusStats ] = useAstarteAPI(astarte.getDevicesStats);
+  const [ interfaces, interfacesStats ] = useAstarteAPI(astarte.getInterfaceNames);
+  const [ triggers, triggersStats ] = useAstarteAPI(astarte.getTriggerNames);
+  const appengineStatus = useAstarteHealth(astarte.getAppengineHealth);
+  const realmManagementStatus = useAstarteHealth(astarte.getRealmManagementHealth);
+  const pairingStatus = useAstarteHealth(astarte.getPairingHealth);
+  let flowStatus = null;
+  if (astarte.config.enableFlowPreview) {
+    flowStatus = useAstarteHealth(astarte.getFlowHealth);
   }
```

Comment:
You are mixing a `null` value with the well defined state machine of `useAstarteHealth`. Why?

Reply:
In this case we have a new status to keep track of. When the configuration says it's not enabled, don't query the Flow health endpoint.
I Refactored the code to make it more readable